fem/libceed/ceedsolvers-atpmg.h

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_CEEDSOLVERS_ATPMG_H
+#define MFEM_CEEDSOLVERS_ATPMG_H
+
+#include "../../config/config.hpp"
+
+#ifdef MFEM_USE_CEED
+
+#include <ceed.h>
+
+/** @brief Take given (high-order) CeedElemRestriction and make a new
+    CeedElemRestriction, which corresponds to a lower-order problem.
+
+    Assumes a Gauss-Lobatto basis and tensor product elements.
+
+    This is a setup routine that operates on the host. */
+int CeedATPMGElemRestriction(int order,
```

Comment:
These functions only use libCEED code, would it make sense to see if it should go in libCEED?